SmartDeviceLink/private/SDLPresentKeyboardOperation.m

```diff
@@ -209,6 +209,12 @@ - (void)sdl_keyboardInputNotification:(SDLRPCNotificationNotification *)notifica
     } else if ([onKeyboard.event isEqualToEnum:SDLKeyboardEventAborted] || [onKeyboard.event isEqualToEnum:SDLKeyboardEventCancelled]) {
         // Notify of abort / cancellation
         [self.keyboardDelegate keyboardDidAbortWithReason:onKeyboard.event];
+    } else if ([onKeyboard.event isEqualToEnum:SDLKeyboardEventInputKeyMaskEnabled] || [onKeyboard.event isEqualToEnum:SDLKeyboardEventInputKeyMaskDisabled]) {
```

Comment:
After checking the proposal, I think that this is missing `Android and iOS changes #4`. Here's how I think that would work:

1. The `SDLPresentKeyboardOperation` and `SDLPresentChoiceSetOperation` needs to handle, in `sdl_updateKeyboardPropertiesWithCompletionHandler`, what happens if the keyboard properties that the developer is trying to use are not supported.

1a. Those operations will need to have the current `SDLWindowCapability` passed to them in the initializer in order to make this check.

2. If the `SDLKeyboardProperties` that the developer is attempting to use uses an unsupported `KeyboardLayout`, the operation will need to fail with an error log.
3. If the `SDLKeyboardProperties` that the developer is attempting to use uses too many `customKeys` for the current layout, we will shorten the amount they send to be the maximum amount and print a warning log.
4. If the `SDLKeyboardProperties` that the developer is attempting to use uses a `customKeys` array that is empty, we will remove the array to `nil`.
5. If the `SDLKeyboardProperties` that the developer is attempting to use uses `maskInputCharacters` when that is not supported on the current layout, we will remove the property to `nil`.
6. If the `SDLKeyboardCapabilities` are not available, use existing behavior of sending whatever the developer set.

Reply:
I implemented the proposed logic in a private category:
`"SDLWindowCapability+ScreenManagerExtensions"`
The method:
`- (SDLKeyboardProperties *__nullable)filterValidKeyboardProperties:(SDLKeyboardProperties *__nullable)inKeyboardProperties;`